### PR TITLE
Add Generic Modals

### PR DESCRIPTION
**Pull request by bipuladh**
Add support for Edit Labels Modal
Add support for Edit Annotations Modal

### CODE REVIEW

File: packages/shared/modals/EditAnnotations.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import { K8sKind } from '@openshift-console/dynamic-plugin-sdk/lib/api/common-types';
+import * as _ from 'lodash-es';
```
Comment (SanjalKatiyar) [inline packages/shared/modals/EditAnnotations.tsx:4]:
```suggestion
import * as _ from 'lodash';
```
not sure the reason for this, but IIC I got this suggestion on one of my PR as well...

File: packages/shared/modals/EditAnnotations.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+export const AnnotationsModal: React.FC<AnnotationsModalProps> =
+  withHandlePromise((props: AnnotationsModalProps) => {
+    const { closeModal, inProgress, errorMessage: promiseError } = props;
+    const [tags, setTags] = React.useState(
+      _.isEmpty(props.tags) ? [['', '']] : _.toPairs(props.tags)
+    );
+    const [errorMessage, setErrorMessage] = React.useState(props.errorMessage);
+
+    const { t } = useTranslation();
```
Comment (SanjalKatiyar) [inline packages/shared/modals/EditAnnotations.tsx:57]: u need to add a plugin namespace

File: packages/shared/modals/EditAnnotations.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      e.preventDefault();
+
+      // We just throw away any rows where the key is blank
+      const usedTags = _.reject(tags, (tag) =>
+        _.isEmpty(tag[NameValueEditorPair.Name])
+      );
+
+      const keys = usedTags.map((tag) => tag[NameValueEditorPair.Name]);
+      if (_.uniq(keys).length !== keys.length) {
+        setErrorMessage(t('public~Duplicate keys found.'));
```
Comment (SanjalKatiyar) [inline packages/shared/modals/EditAnnotations.tsx:69]: incorrect namespace

File: packages/shared/utils/NameValueEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  toolTip,
+  valueString,
+  alwaysAllowRemove,
+}) => {
+  const { t } = useTranslation('plugin__odf-console');
+  const [node, setNode] = React.useState(null);
+  const deleteIcon = (
+    <>
+      <MinusCircleIcon className="pairs-list__side-btn pairs-list__delete-icon" />
+      <span className="sr-only">{t('public~Delete')}</span>
```
Comment (SanjalKatiyar) [inline packages/shared/utils/NameValueEditor.tsx:90]: incorrect namespace

File: packages/shared/utils/NameValueEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    </>
+  );
+  const dragButton = (
+    <div>
+      <Button
+        type="button"
+        className="pairs-list__action-icon"
+        isDisabled={disableReorder}
+        variant="plain"
+        aria-label={t('public~Drag to reorder')}
```
Comment (SanjalKatiyar) [inline packages/shared/utils/NameValueEditor.tsx:100]: namespace

**Comment by bipuladh** [comment]
Screenshot from 2022-03-17 14-20-19
Screenshot from 2022-03-17 14-20-03
Tested everything works fine updating,removing,adding labels and annotations. 
Some CSS related work needs to be done to achieve uniformity with Console. We can do that after migrating other modals required for ODF.

**Pull request merged**